File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        l1=list(s)
        l=0
        r=len(l1)-1
        while l<r:
            if l1[l].lower() not in "aeiou":
                l+=1
                continue
            if l1[r].lower() not in "aeiou":
                r-=1
                continue
            temp=l1[l]
            l1[l]=l1[r]
            l1[r]=temp
            l+=1
            r-=1
        return "".join(l1)

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.java
class Solution {
    public String reverseVowels(String s) {
        char[] arr1=s.toCharArray();
        int l=0;
        int r=arr1.length-1;
        while(l<r){
            if("aeiou".indexOf(Character.toLowerCase(arr1[l]))==-1){
                l++;
                continue;
            }
            if("aeiou".indexOf(Character.toLowerCase(arr1[r]))==-1){
                r--;
                continue;
            }
            char temp =arr1[l];
            arr1[l]=arr1[r];
            arr1[r]=temp;
            l++;
            r--;
        }
        return new String(arr1);
    }
}